src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/ReflectionMarker.cs

```diff
@@ -198,6 +198,9 @@ internal void MarkStaticConstructor(in MessageOrigin origin, TypeDesc type, stri
             if (!_enabled)
                 return;
 
+            if (type.HasStaticConstructor)
+                CheckAndWarnOnReflectionAccess(origin, type.GetStaticConstructor());
```

Comment:
Will this cause two warnings to be generated? If I understand this correctly, this will generate a warning for reflection access to cctor, and then we get another one when we generate the method body, because this attribute on a cctor is always an error.

Reply:
It actually won't - we intentionally ignore the `Requires*` attributes on static cctor for generating warnings (basically relying on the fact that they will produce a warning just due to their presence, without actual callsite).
https://github.com/dotnet/runtime/blob/24fa97ab99b4a63f638f63a90ae83c4b80772114/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/DiagnosticUtilities.cs#L100
https://github.com/dotnet/runtime/blob/24fa97ab99b4a63f638f63a90ae83c4b80772114/src/tools/illink/src/linker/Linker/Annotations.cs#L609
https://github.com/dotnet/runtime/blob/24fa97ab99b4a63f638f63a90ae83c4b80772114/src/tools/illink/src/ILLink.RoslynAnalyzer/RequiresISymbolExtensions.cs#L60

Reply:
I'm adding a test for this behavior as well.